### PR TITLE
BLUEVIEW: use telodyne sdk instead of custom package

### PR DESCRIPTION
**Pull request by kev-the-dev**
addresses #11

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -521,8 +521,8 @@ sudo pip install -q -U tqdm
 # The BlueView SDK for the Teledyne imaging sonar
 if [[ ! -z "$BVTSDK_PASSWORD" ]]; then
 	instlog "Decrypting and installing the BlueView SDK"
-	cd $CATKIN_DIR/src
-	curl -s https://raw.githubusercontent.com/uf-mil/installer/master/bvtsdk.tar.gz.enc | \
+	cd $CATKIN_DIR/src/mil_common/drivers/mil_blueview_driver/
+	curl -s https://raw.githubusercontent.com/uf-mil/installer/master/libbvtsdk.tar.gz.enc | \
 	openssl enc -aes-256-cbc -d -pass file:<(echo -n $BVTSDK_PASSWORD) | tar -xpz
 	if [[ ! -d $CATKIN_DIR/src/bvtsdk ]]; then
```
Comment (sentree) [inline install.sh]: Why did you not change this? Be sure you look over everything relating to the SDK.

File: install.sh [hunk 2]
```diff
@@ -521,8 +521,8 @@ sudo pip install -q -U tqdm
 # The BlueView SDK for the Teledyne imaging sonar
 if [[ ! -z "$BVTSDK_PASSWORD" ]]; then
 	instlog "Decrypting and installing the BlueView SDK"
-	cd $CATKIN_DIR/src
-	curl -s https://raw.githubusercontent.com/uf-mil/installer/master/bvtsdk.tar.gz.enc | \
+	cd $CATKIN_DIR/src/mil_common/drivers/mil_blueview_driver/
+	curl -s https://raw.githubusercontent.com/uf-mil/installer/master/libbvtsdk.tar.gz.enc | \
 	openssl enc -aes-256-cbc -d -pass file:<(echo -n $BVTSDK_PASSWORD) | tar -xpz
```
Comment (sentree) [inline install.sh]: Am I correct in assuming that you used the same password and tested this decryption line verbatim?

File: install.sh [hunk 3]
```diff
@@ -521,8 +521,8 @@ sudo pip install -q -U tqdm
 # The BlueView SDK for the Teledyne imaging sonar
 if [[ ! -z "$BVTSDK_PASSWORD" ]]; then
 	instlog "Decrypting and installing the BlueView SDK"
-	cd $CATKIN_DIR/src
-	curl -s https://raw.githubusercontent.com/uf-mil/installer/master/bvtsdk.tar.gz.enc | \
+	cd $CATKIN_DIR/src/mil_common/drivers/mil_blueview_driver/
+	curl -s https://raw.githubusercontent.com/uf-mil/installer/master/libbvtsdk.tar.gz.enc | \
```
Comment (sentree) [inline install.sh]: Why change the name of the encrypted archive, but not the directory it extracts to? I think the name should stay as is.

File: install.sh [hunk 4]
```diff
@@ -521,8 +521,8 @@ sudo pip install -q -U tqdm
 # The BlueView SDK for the Teledyne imaging sonar
 if [[ ! -z "$BVTSDK_PASSWORD" ]]; then
 	instlog "Decrypting and installing the BlueView SDK"
-	cd $CATKIN_DIR/src
-	curl -s https://raw.githubusercontent.com/uf-mil/installer/master/bvtsdk.tar.gz.enc | \
+	cd $CATKIN_DIR/src/mil_common/drivers/mil_blueview_driver/
```
Comment (sentree) [inline install.sh]: No trailing slash, here. The use of `cd` implies that it is a directory and it is not convention for the file.

**Review by sentree — CHANGES_REQUESTED** [review]
This version of the script would clearly fail every time due to my last comment in this review. Any changed logic in the script needs to be tested in a bare chroot or virtual machine prior to merge (this means each condition). A docker rebuild will also have to be attempted prior to merge.

See #11 for other things that have to be done to get building to work.

**Comment by kev-the-dev** [comment]
Updated so it will extract tarball to ```~/.mil/bvtsdk```. Then it will symbolic link this to the correct location in the mil_blueview_driver, or if on docker just leave it there. Than the Jenkinsfile for SubjuGator / mil_common will link to to the mil_blueview_driver after they pull mil_common.

**Comment by kev-the-dev** [comment]
Let's try rebuilding the docker image today using my install script branch then manually pull my mil_common branch and build it like Jenkins would

**Comment by kev-the-dev** [comment]
Just tested on my own system. Does correctly download, decrypt, link to driver directory, and build. Just gatta check docker now.

File: install.sh [hunk 5]
```diff
@@ -53,6 +53,8 @@ CATKIN_DIR=~/mil_ws
 MIL_CONFIG_DIR=~/.mil
 BASHRC_FILE=~/.bashrc
 MILRC_FILE=$MIL_CONFIG_DIR/milrc
+BVTSDK_DIR=$MIL_CONFIG_DIR
+BVTSDK_DIR_COMPLETE=$BVTSDK_DIR/bvtsdk
```
Comment (sentree) [inline install.sh]: Why create a new variable just to mirror `MIL_CONFIG_DIR`? I think it world be better to just use `BVTSDK_DIR` for what `BVTSDK_DIR_COMPLETE` is being used for and then remove `BVTSDK_DIR_COMPLETE`.

File: install.sh [hunk 6]
```diff
@@ -521,12 +523,14 @@ sudo pip install -q -U tqdm
 # The BlueView SDK for the Teledyne imaging sonar
 if [[ ! -z "$BVTSDK_PASSWORD" ]]; then
 	instlog "Decrypting and installing the BlueView SDK"
-	cd $CATKIN_DIR/src
-	curl -s https://raw.githubusercontent.com/uf-mil/installer/master/bvtsdk.tar.gz.enc | \
+	cd $BVTSDK_DIR
+	curl -s https://raw.githubusercontent.com/uf-mil/installer/master/libbvtsdk.tar.gz.enc | \
 	openssl enc -aes-256-cbc -d -pass file:<(echo -n $BVTSDK_PASSWORD) | tar -xpz
```
Comment (sentree) [inline install.sh]: Am I correct in assuming that you used the same password and tested this decryption line verbatim?

File: install.sh [hunk 7]
```diff
@@ -521,12 +523,14 @@ sudo pip install -q -U tqdm
 # The BlueView SDK for the Teledyne imaging sonar
 if [[ ! -z "$BVTSDK_PASSWORD" ]]; then
 	instlog "Decrypting and installing the BlueView SDK"
-	cd $CATKIN_DIR/src
-	curl -s https://raw.githubusercontent.com/uf-mil/installer/master/bvtsdk.tar.gz.enc | \
+	cd $BVTSDK_DIR
+	curl -s https://raw.githubusercontent.com/uf-mil/installer/master/libbvtsdk.tar.gz.enc | \
```
Comment (sentree) [inline install.sh]: Why change the name of the encrypted archive, but not the directory it extracts to?

File: install.sh [hunk 8]
```diff
@@ -521,12 +523,14 @@ sudo pip install -q -U tqdm
...
 	openssl enc -aes-256-cbc -d -pass file:<(echo -n $BVTSDK_PASSWORD) | tar -xpz
-	if [[ ! -d $CATKIN_DIR/src/bvtsdk ]]; then
+	if [[ ! -d $BVTSDK_DIR_COMPLETE ]]; then
 		instwarn "Terminating installation due to incorrect password for the BlueView SDK"
 		exit 1
+	elif [[ "$DOCKER" != "true" ]]; then
+		ln -s $BVTSDK_DIR_COMPLETE $CATKIN_DIR/src/mil_common/drivers/mil_blueview_driver/bvtsdk
```
Comment (sentree) [inline install.sh]: Try to comment this block a bit more, it could be confusing for people not involved in developing the CI process.

**Review by sentree — CHANGES_REQUESTED** [review]
Looks a lot better now, I will test this with Docker once you make the changes. Be sure to mirror your changes in `user_install.sh` as those files need to stay in sync.

File: install.sh (hunk 7 above)
Comment (kev-the-dev) [reply install.sh]: I'm just mirroring the name the sdk comes as. It is distrubted as libbvtsdk.tar.gz and contains a bvtsdk directory.

File: install.sh (hunk 6 above)
Comment (kev-the-dev) [reply install.sh]: Yee it works

File: install.sh (hunk 5 above)
Comment (kev-the-dev) [reply install.sh]: I like it being it's own variable so it's easy to change one line (56) if we decide to put BVTSDK somewhere else like .local or something.

**Review by sentree — APPROVED** [review]
Looks good.

**Pull request merged**